### DISCUSSION TITLE
Add ability to see records not scoped to the current collection

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5723] [Feature Requests]
This is an extension of https://github.com/specify/specify7/issues/1886

I know that we have discussed this issue in the past, but this is something we should discuss as a team and come up with an answer for. 

**Requested By:** Willem at SAIAB

> I have a Tissue Collection, which is a collection in the Ichthyology Discipline. As you know I have other disciplines and collections, all of which are related to collectionobject records in the Tissue Collection (collectionrelationship).
> 
> While this has worked well, I have come across a constraint in that some information does not display when I do a query starting from the right side (Tissue Collection) and including fields from the left side.
> 
> The fields that don’t display are the taxon levels (Kingdom, Phylum etc.) for all the collections that are not in the Ichthyology Discipline. They display for the Fish Collection because it is in the same discipline as the Tissue Collection.

**Comment by grantfitzsimmons** [comment]
May be assisted by https://github.com/specify/specify7/issues/2903

**Comment by maxpatiiuk** [comment]
@grantfitzsimmons my short term plan for this. need your feedback:

We extend the table scoping detection to properly identify the scope for all tables (AccessionAgents, Borrow, etc)
If record is scoped to collection you don't have access to, you won't be able to edit just like now
Otherwise, the form will be visible, but with a red message at the top saying you better look at this record in a different collection.
If I am correct, the risk of them editing the record from inside a different collection is not too horrible (though not great) - worst case: web links may not work, wrong pick lists may be used, wrong form definition is used, wrong field captions are used.
Of course, the code would have to be modified to check for permissions from the other collection when editing the record from the other collection.
This change will be the first step toward fixing https://github.com/specify/specify7/issues/1886

**Comment by grantfitzsimmons** [comment]
> We extend the table scoping detection to properly identify the scope for all tables (AccessionAgents, Borrow, etc)
> If record is scoped to collection you don't have access to, you won't be able to edit just like now
> Otherwise, the form will be visible, but with a red message at the top saying you better look at this record in a different collection.
> If I am correct, the risk of them editing the record from inside a different collection is not too horrible (though not great) - worst case: web links may not work, wrong pick lists may be used, wrong form definition is used, wrong field captions are used.
> Of course, the code would have to be modified to check for permissions from the other collection when editing the record from the other collection.
> This change will be the first step toward fixing #1886

I like your plan. I wonder if it's too late for you to implement it :wink: :joy:

**Comment by grantfitzsimmons** [comment]
From @realVinayak:

> **Describe the bug**
> Consider this query over collecting events. I'm currently logged in KU Fish Teaching collection. The query builder is returning collecting events which have collection objects from a different collection. In the screenshot, the collection object returned is from KU Fish Voucher collection - this is fine, and not an issue. 
> 
> However, if you were to make any changes to the form, then it is not letting me save it since it requires catalog number to be of 5 digits (in KU Fish Teaching Collection). Since the catalog number need to be of more digits in the KU Fish Voucher Collection), the form throws an error saying the catalog number doesn't match the required format.
> 
> The collecting event - https://coldfish-edge-2.test.specifysystems.org/specify/view/collectingevent/4/
> 
> The bug screenshot
> ![image](https://user-images.githubusercontent.com/61122018/228005146-604b00f7-c35b-42ae-8f60-9d0167642ea3.png)
> 
>
